app/broadcast_message/rest.py

```diff
@@ -190,32 +190,42 @@ def update_broadcast_message_status(service_id, broadcast_message_id):
 
 def _create_broadcast_event(broadcast_message):
     """
-    Creates a broadcast event, stores it in the database, and triggers the task to send the CAP XML off
+    If the service is live and the broadcast message is not stubbed, creates a broadcast event, stores it in the
+    database, and triggers the task to send the CAP XML off.
     """
-    msg_types = {
-        BroadcastStatusType.BROADCASTING: BroadcastEventMessageType.ALERT,
-        BroadcastStatusType.CANCELLED: BroadcastEventMessageType.CANCEL,
-    }
-
-    event = BroadcastEvent(
-        service=broadcast_message.service,
-        broadcast_message=broadcast_message,
-        message_type=msg_types[broadcast_message.status],
-        transmitted_content={"body": broadcast_message.content},
-        transmitted_areas=broadcast_message.areas,
-        # TODO: Probably move this somewhere more standalone too and imply that it shouldn't change. Should it include
-        # a service based identifier too? eg "flood-warnings@notifications.service.gov.uk" or similar
-        transmitted_sender='notifications.service.gov.uk',
-
-        # TODO: Should this be set to now? Or the original starts_at?
-        transmitted_starts_at=broadcast_message.starts_at,
-        transmitted_finishes_at=broadcast_message.finishes_at,
-    )
+    service = broadcast_message.service
+
+    if not broadcast_message.stubbed and not service.restricted:
```

Comment:
Minor: we could invert this and return early (after logging), to avoid some of the nesting. Then it would look like:

```
if stubbed or restricted:
  log_error
  return

rest_of_method
```

Reply:
We don't want to log an error if the broadcast message is stubbed _and_ the service is restricted (this is the scenario where a trial mode service can approve their own broadcasts) - only if there's a mismatch between the stubbed and restricted values. This means the logic would need to look slightly more complicated if changed:
```python
if stubbed or restricted:
    if stubbed != restricted:
        log_error
  return

rest_of_method
```

Reply:
Oh yes, thanks for explaining. In that case I don't think it makes much difference either way, so let's leave it as-is 👍.